pages/_app.js

```diff
@@ -1,42 +1,57 @@
 import GlobalStyle from "@/styles";
-import useLocalStorageState from "use-local-storage-state";
 import "react-toastify/dist/ReactToastify.css";
 import { StyledToastContainer } from "@/components/Toast";
 import { useState } from "react";
 import Layout from "@/components/Layout";
 import Fuse from "fuse.js";
-import useSWR, { SWRConfig } from "swr";
+import useSWR, { mutate, SWRConfig } from "swr";
+import { getSession, SessionProvider } from "next-auth/react";
+import useLocalStorageState from "use-local-storage-state";
 
 const fetcher = (url) => fetch(url).then((res) => res.json());
 
-export default function App({ Component, pageProps }) {
+export default function App({
+  Component,
+  pageProps: { session, ...pageProps },
+}) {
   const {
     data: activityData,
     error,
     isLoading,
   } = useSWR("/api/activities", fetcher);
 
-  const [favoriteActivitiesList, setFavoriteActivitiesList] =
-    useLocalStorageState("favorites", {
-      defaultValue: [],
-    });
+  const { data: userData } = useSWR("/api/users", fetcher);
+
   const [randomActivity, setRandomActivity] = useLocalStorageState(
     "randomActivity",
     { defaultValue: null }
   );
   const [selectedCategory, setSelectedCategory] = useState("");
   const [searchTerm, setSearchTerm] = useState("");
 
-  function handleToggleFavorite(id) {
-    setFavoriteActivitiesList((prevFavorites) => {
-      const isFavorite = prevFavorites.some((activity) => activity._id === id);
-      if (isFavorite) {
-        return prevFavorites.filter((activity) => activity._id !== id);
-      } else {
-        const activity = activityData?.find((activity) => activity._id === id);
-        return [...prevFavorites, { ...activity, isFavorite: true }];
-      }
+  async function handleToggleFavorite(id) {
+    const session = await getSession();
```

Comment:
The getSession() helper function is intended to be used outside of a react context. Better use the useSession hook here.